### DISCUSSION TITLE
strongswan 6.0.0 beta6 Question

### DISCUSSION THREAD

**Discussion by rikozz** [strongswan/strongswan#2359] [General]
hello. I am testing ikev2 multiple key exchange [related to RFC9370] with strongswan-6.0.0beta6 (with liboqs).
I have one question.
It looks like there is no additional(multiple) key negotiation for ipsec sa (child sa) at the time the IKE_AUTH message processing is completed. Is that correct?
It seems that there is multiple key exchange only in the rekey operation for the created child sa.
If you look at the src/libcharon/sa/ikev2/tasks/child_create.c file, in the case of IKE_AUTH, the proposal selection logic is treated as no_ke = TRUE, so the expectation seems to be correct. 
Can someone please give me an answer.

**Comment by tobiasbrunner** [comment]
You already referenced the RFC, why not read it?

**Reply by rikozz** [reply]
I'm sorry. There was a problem with my question. As I focused on testing, I forgot the contents of the RFC. Why is RFC9370 designed not to perform multiple key exchange in the first IPSec SA?
Do you think that a sufficient level of security has been achieved through the multiple key exchanges that occurred during the first IKE SA negotiation?

**Reply by tobiasbrunner** [reply]
There is never a separate key exchange for the the CHILD_SA that's created with IKE_AUTH, its keys are always derived from the IKE_SA's key material (which is potentially created with multiple key exchanges via IKE_INTERMEDIATE). Changing that would have become a lot more complicated. If you don't want that, use childless IKE_SA initiation (RFC 6023) so the first CHILD_SA is created in a separate CREATE_CHILD_SA exchange (that's the same with classical DH algorithms if you want to enforce PFS even for the first CHILD_SA). However, some have argued that protecting the IKE_SA with PQC key exchange algorithms is already enough and the CHILD_SAs don't even need additional protection as their keys are either directly derived from the IKE_SA's key material, or their classical key exchange (if PFS is used, which is optional anyway) is protected by the IKE_SA.

**Reply by rikozz** [reply]
Thank you very much for the detailed answer. 
It was very helpful.
